micrometer-core/src/main/java/io/micrometer/core/instrument/internal/TimedExecutor.java

```diff
@@ -15,28 +15,32 @@
  */
 package io.micrometer.core.instrument.internal;
 
+import java.util.concurrent.Executor;
+
 import io.micrometer.core.instrument.MeterRegistry;
 import io.micrometer.core.instrument.Tag;
 import io.micrometer.core.instrument.Tags;
 import io.micrometer.core.instrument.Timer;
 
-import java.util.concurrent.Executor;
-
 /**
  * An {@link Executor} that is timed
  */
 public class TimedExecutor implements Executor {
+    private final MeterRegistry registry;
     private final Executor delegate;
-    private final Timer timer;
+    private final Timer executionTimer;
+    private final Timer idleTimer;
 
     public TimedExecutor(MeterRegistry registry, Executor delegate, String executorName, Iterable<Tag> tags) {
+        this.registry = registry;
         this.delegate = delegate;
-        this.timer = registry.timer("executor", Tags.concat(tags, "name", executorName));
+        this.executionTimer = registry.timer("executor.execution", Tags.concat(tags, "name", executorName));
```

Comment:
Unfortunately we can't just change the name of the metric as this will break users relying on the name being `executor` as it has been. If we want to correct the inconsistency, we'll need to add another timer that times the same thing with the new name. Then in Micrometer 2.0, we can remove the `executor` one.